include/boost/url/impl/decoded_view.ipp

```diff
@@ -0,0 +1,135 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_IMPL_DECODED_VIEW_IPP
+#define BOOST_URL_IMPL_DECODED_VIEW_IPP
+
+#include <boost/url/decoded_view.hpp>
+#include <ostream>
+
+namespace boost {
+namespace urls {
+
+auto
+decoded_view::iterator::
+operator*() const noexcept ->
+    reference
+{
+    if (plus_to_space_ &&
+        *pos_ == '+')
+        return ' ';
+    if (*pos_ != '%')
+        return *pos_;
+    char d0;
+    grammar::hexdig_value(pos_[1], d0);
```

Comment:
is there a reason why d0 cannot be unsigned from the start? Seems more sensible to me. And - less typing!

Reply:
`hexdig_value` returns -1 on error

Reply:
actually `hexdig_value` returns a `bool`.